### ISSUE TITLE
The example in `Diagrams.SVG.ReadSVG` doesn't type-check.

### ISSUE THREAD

**Issue by hansroland** [diagrams/diagrams-input#14]
The main part of the example is:

````
main = do
   diagramFromSVG  <- readSVGFile "svgs/web.svg"
   mainWith $ diagramFromSVG
````

The error is:

````
Main.hs:12:22: error:
    • Couldn't match type ‘N b0’ with ‘Place’
        arising from a use of ‘readSVGFile’
      The type variable ‘b0’ is ambiguous
    • In a stmt of a 'do' block:
        diagramFromSVG <- readSVGFile "svgs/web.svg"
      In the expression:
        do diagramFromSVG <- readSVGFile "svgs/web.svg"
           mainWith $ diagramFromSVG
      In an equation for ‘main’:
          main
            = do diagramFromSVG <- readSVGFile "svgs/web.svg"
                 mainWith $ diagramFromSVG
````

The function `readSVGFile`returns an `Either String (Diagram B)` but
the function `mainWith` expects a plain `Diagram B`.

A working example is:

````
{-# LANGUAGE OverloadedStrings, ScopedTypeVariables #-}

module Main where
import Diagrams.SVG.ReadSVG
import Diagrams.Prelude
import Diagrams.Backend.SVG.CmdLine
import System.Environment
import Filesystem.Path.CurrentOS
import Diagrams.SVG.Attributes (PreserveAR(..), AlignSVG(..), Place(..), MeetOrSlice(..))

main = do
   diagramFromSVG :: Either String (Diagram B) <- readSVGFile "svgs/web.svg"
   case diagramFromSVG of
    Left msg      -> error ("readSVGFile returned " <> msg)
    Right diagram -> mainWith $ diagram
````

It needs a type hint for `diagramFromSVG`.

Examples from the documentation of a library should work without any changes.

Verions used:

* GHC: The Glorious Glasgow Haskell Compilation System, version 9.2.2
* diagrams-input: 0.1

**Comment by hansroland** [comment]
If you want, I can create a PR.

**Comment by byorgey** [comment]
@hansroland thanks for the report!  Yes, that seems like a mistake.  Perhaps the type of `readSVGFile` changed at some point, I am not sure.  A PR would be greatly appreciated!

**Comment by byorgey** [comment]
Closed by #16 .
